sys/sys/signal.h

```diff
@@ -219,78 +202,56 @@ union sigval32 {
 };
 #endif
 
-#ifdef _KERNEL
-typedef union {
-	int		sival_int;
-	void		*sival_ptr_native;
-#if (defined(_KERNEL) && defined(__LP64__))
-	uint32_t	sival_ptr32;
-#endif
-#if __has_feature(capabilities)
-	void * /* __ptr64 */ sival_ptr64;
-	void * __capability sival_ptr_c;
+#if defined(_WANT_LWPINFO64) || (defined(_KERNEL) && defined(COMPAT_FREEBSD64))
+union sigval64 {
+	int	sival_int;
+	uint64_t sival_ptr;
+};
 #endif
-} ksigval_union;
+
+#if defined(_KERNEL) && __has_feature(capabilities)
```

Comment:
This should probably be `defined(COMPAT_CHERIABI)` instead of `__has_feature()`.  Makes it easier to axe when the time comes.

Reply:
Done